### PR TITLE
Add Connections, Start Testing Backend

### PR DESCRIPTION
**Pull request by beeequeue**

### CODE REVIEW

File: packages/server/src/modules/google/google.auth.test.ts [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+  const setCookies = response.header['set-cookie'] as string[]
+  const lastCookie = setCookies[setCookies.length - 1]
+
+  const token = /token=([\w\d-])+;/
+    .exec(lastCookie)![0]
+    .slice(6, -1)
+
+  expect(token).not.toBeNull()
+  expect(token).toMatch(
+    /^[0-9A-F]{8}-[0-9A-F]{4}-4[0-9A-F]{3}-[89AB][0-9A-F]{3}-[0-9A-F]{12}$/i,
```
Comment (peoplenarthax) [inline packages/server/src/modules/google/google.auth.test.ts:54]: do we really need to test the format of the token?

File: packages/server/src/modules/google/google.routes.ts [hunk 2]
```diff
@@ -1,11 +1,31 @@
...
+} from '@/modules/connection/connection.model'
 import { isNil } from '@/utils'
+import { Google } from './google.lib'
+
+const redirectToFailure = (
+  res: Response,
+  code: AuthErrorCode,
+  extraParams: { [key: string]: string } = {},
+) => {
+  const url = new URL('http://frontend.url/connect/failed')
```
Comment (peoplenarthax) [inline packages/server/src/modules/google/google.routes.ts:18]: To Do: Add correct url

File: packages/server/src/modules/user/user.model.ts [hunk 3]
```diff
@@ -1,41 +1,68 @@
...
   @Field(() => ID, { nullable: true })
-  public mainConnectionUuid?: string
+  public mainConnectionUuid!: string | null
+
+  constructor(options: UserConstructor) {
+    super()
+
+    if (isNil(options)) options = {} as any
+
+    this.uuid = options.uuid || uuid()
```
Comment (peoplenarthax) [inline packages/server/src/modules/user/user.model.ts]:
```suggestion
    this.uuid = options.uuid ?? uuid()
```

File: packages/server/src/modules/user/user.model.ts [hunk 4]
```diff
@@ -1,41 +1,68 @@
...
+      type: options.type,
+      userUuid: this.uuid,
+      email: options.email,
+      serviceId: options.serviceId,
+      image: options.image,
+    }).save()
 
-    // TODO: Filter keys haha
-    return Object.assign(user, parameters)
+    this.mainConnectionUuid = this.mainConnectionUuid || connection.uuid
```
Comment (peoplenarthax) [inline packages/server/src/modules/user/user.model.ts]:
```suggestion
    this.mainConnectionUuid = this.mainConnectionUuid ?? connection.uuid
```

File: packages/server/src/modules/connection/connection.model.ts [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+  @Column()
+  @Field()
+  public image: string
+
+  constructor(options: ConnectionConstructor) {
+    super()
+
+    if (isNil(options)) options = {} as any
+
+    this.uuid = options.uuid || uuid()
```
Comment (peoplenarthax) [inline packages/server/src/modules/connection/connection.model.ts]:
```suggestion
    this.uuid = options.uuid ?? uuid()
```

**Comment by peoplenarthax** [comment]
looking good, in a way it feels like we are testing Google, but I think is nice to have those test, more TDD ? 

I just added some suggestions for nullish coallescing and then I approve and move to the next PR

File: packages/server/src/modules/google/google.routes.ts (hunk 2 above)
Comment (beeequeue) [reply packages/server/src/modules/google/google.routes.ts:18]: Need to add a `FRONTEND_URL` var so we know where to send users back

Comment (beeequeue) [reply packages/server/src/modules/google/google.routes.ts:18]: Or `redirectTo=https://`

File: packages/server/src/modules/google/google.auth.test.ts (hunk 1 above)
Comment (beeequeue) [reply packages/server/src/modules/google/google.auth.test.ts:54]: Not really

Comment (peoplenarthax) [reply packages/server/src/modules/google/google.auth.test.ts:54]: It also does not hurt

File: packages/server/src/modules/google/google.routes.ts (hunk 2 above)
Comment (peoplenarthax) [reply packages/server/src/modules/google/google.routes.ts:18]: We can do it in other PR tho, it is good enough in this one

**Review by peoplenarthax — APPROVED** [review]
Lekert!

**Pull request merged**